.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,95 @@
+name: "CI/CD"
+
+on:
+  push:
+    branches:
+      - main
+      - develop
+      - release/*
+  workflow_dispatch:
+
+jobs:
+  publish:
+    name: "Publish on expo.dev"
+    uses: ./.github/workflows/publish.yml
+
+  build-android:
+    name: "Build Android APK"
+    uses: ./.github/workflows/build-android.yml
+
+  build-ios:
+    name: "Build iOS App"
+    uses: ./.github/workflows/build-ios.yml
+
+  check-release-version:
+    name: "Check release version"
+    if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/heads/release/')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup repo
+        uses: actions/checkout@v2
+        with:
+          # actions/checkout does a shallow clone by default, which doesn't pull any tags
+          fetch-depth: 0
+
+      - name: Get current package version
+        id: extract_version
+        uses: Saionaro/extract-package-version@v1.0.6
+
+      - name: Check that package version and release version match
+        if: startsWith(github.ref, 'refs/heads/release/')
+        run: |
+          ref_version="${GITHUB_REF#refs/heads/release/}"
+          if [[ "$ref_version" != "${{ steps.extract_version.outputs.version }}" ]]; then
+            echo "::error title=Version Mismatch::Version '$ref_version' from branch name does not match package version '${{ steps.extract_version.outputs.version }}'. Did you bump the package version?"
+            exit 1
+          fi
+
+      - name: Check that no tag for the current version already exists
+        run: |
+          git tag -l
+          if git tag -l | grep -Fxq "v${{ steps.extract_version.outputs.version }}"; then
+            echo "::error title=v${{ steps.extract_version.outputs.version }} already exists::The release tag 'v${{ steps.extract_version.outputs.version }}' already exists."
+            exit 1
+          fi
+
+  release:
+    name: "Release"
+    if: github.ref == 'refs/heads/main'
+    needs: [ publish, build-android, build-ios, check-release-version ]
```

Comment:
Werden  hier `build-android` und `build-ios` durch `needs:` getriggered? Beide Workflows werden soweit ich weiß ansonsten durch manuellen dispatch gestartet.

Reply:
Needs gibt an, dass der release Job nur läuft, nachdem diese vier erfolgreich durchgelaufen sind. `build-android` referenziert also den _Job_ `build-android` in diesem Workflow, und der wiederum ruft den _Workflow_ `build-android` auf. Ich habe das so gemacht, um eben hier `needs` verwenden zu können und gleichzeitig die Möglichkeit zu haben, nur den Android- oder iOS-Build manuell zu triggern.

Reply:
Alles klar habe ich mir schon gedacht. Obwohl wir jetzt public sind und kein gh runner Minuten Limit haben, finde ich das die beste Lösung. Der atifact Speicher bedankt sich sicher auch :)